File: flutter_app/lib/models/flashcard.dart
class Flashcard {
  final String
  id;
  final String
  front;
  final String
  back;
  final String
  subject;
  final int
  difficulty;
  final DateTime
  nextReview;
  final int
  reviewCount;

  Flashcard({
    required this.id,
    required this.front,
    required this.back,
    required this.subject,
    required this.difficulty,
    required this.nextReview,
    required this.reviewCount,
  });

  factory Flashcard.fromJson(
    Map<
      String,
      dynamic
    >
    json,
  ) {
    return Flashcard(
      id: json['_id'],
      front: json['front'],
      back: json['back'],
      subject: json['subject'],
      difficulty:
          json['difficulty'] ??
          0,
      nextReview: DateTime.parse(
        json['nextReview'],
      ),
      reviewCount:
          json['reviewCount'] ??
          0,
    );
  }
}

File: backend/src/models/Flashcard.js
const mongoose = require('mongoose');

const flashcardSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  front: {
    type: String,
    required: true,
  },
  back: {
    type: String,
    required: true,
  },
  subject: {
    type: String,
    required: true,
  },
  difficulty: {
    type: Number,
    default: 0, // Spaced repetition difficulty
  },
  nextReview: {
    type: Date,
    default: Date.now,
  },
  reviewCount: {
    type: Number,
    default: 0,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Flashcard', flashcardSchema);
